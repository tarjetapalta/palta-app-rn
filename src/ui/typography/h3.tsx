import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H3: FunctionComponent<TextProps> = ({
  /** Test id for test */
  testID = 'H3',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="l"
    weight="medium"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
