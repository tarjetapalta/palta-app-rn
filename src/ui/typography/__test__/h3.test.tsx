import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { H3 } from '../h3';

describe('h3 component', () => {
  it('basic H3 with color "text", emphasis "high" and testID "H3"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H3>Some text</H3>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic H3 with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H3 color="primary" emphasis="medium">
            Some text
          </H3>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H3 testID="custom testID">Some text</H3>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
