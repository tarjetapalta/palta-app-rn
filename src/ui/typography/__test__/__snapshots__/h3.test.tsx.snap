// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h3 component basic H3 with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="l"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Medium",
        "fontSize": 18,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  testID="H3"
  weight="medium"
>
  Some text
</Text>
`;

exports[`h3 component basic H3 with color "text", emphasis "high" and testID "H3" 1`] = `
<Text
  numberOfLines={0}
  size="l"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Medium",
        "fontSize": 18,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="H3"
  weight="medium"
>
  Some text
</Text>
`;

exports[`h3 component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="l"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Medium",
        "fontSize": 18,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
  weight="medium"
>
  Some text
</Text>
`;
